nomic/embed.py

```diff
@@ -61,18 +76,104 @@ def text_api_request(texts: List[str], model: str, task_type: str, dimensionalit
         raise Exception((response.status_code, response.text))
 
 
-def text(texts: List[str], model: str = "nomic-embed-text-v1", task_type: str = "search_document", dimensionality: int = None, long_text_mode: str = "truncate"):
+@overload
+def text(
+    texts: list[str],
+    *,
+    model: str = ...,
+    task_type: str = ...,
+    dimensionality: int | None = ...,
+    long_text_mode: str = ...,
+    inference_mode: Literal["atlas"] = ...,
+) -> dict[str, Any]: ...
+@overload
+def text(
+    texts: list[str],
+    *,
+    model: str = ...,
+    task_type: str = ...,
+    dimensionality: int | None = ...,
+    long_text_mode: str = ...,
+    inference_mode: Literal["local"],
+    **kwargs: Any,
+) -> dict[str, Any]: ...
+@overload
+def text(
+    texts: list[str],
+    *,
+    model: str = ...,
+    task_type: str = ...,
+    dimensionality: int | None = ...,
+    long_text_mode: str = ...,
+    inference_mode: str,
+    **kwargs: Any,
+) -> dict[str, Any]: ...
+
+
+def text(
+    texts: list[str],
+    *,
+    model: str = "nomic-embed-text-v1",
+    task_type: str = "search_document",
+    dimensionality: int | None = None,
+    long_text_mode: str = "truncate",
+    inference_mode: str = "remote",
+    device: str | None = None,
+    **kwargs: Any,
+) -> dict[str, Any]:
     """
     Generates embeddings for the given text.
 
     Args:
-        texts: the text to embed
-        model: the model to use when embedding
-        task_type: the task type to use when embedding. One of `search_query`, `search_document`, `classification`, `clustering`
+        texts: The text to embed.
+        model: The model to use when embedding.
+        task_type: The task type to use when embedding. One of `search_query`, `search_document`, `classification`, `clustering`.
+        dimensionality: The embedding dimension, for use with Matryoshka-capable models. Defaults to full-size.
+        long_text_mode: How to handle texts longer than the model can accept. One of `mean` or `truncate`.
+        inference_mode: How to generate embeddings. One of `remote` or `local` (Embed4All). Defaults to `remote`.
+        device: The device to use for local embeddings. Defaults to CPU, or Metal on Apple Silicon. It can be set to:
```

Comment:
should we log or print something that we ignore `device` when `inference_mode` is `atlas`?